hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -119,12 +118,14 @@
 	private final UniqTokenViewsManager uniqTokenViewsManager;
 	private final GlobalDynamicProperties properties;
 	private final SideEffectsTracker sideEffectsTracker;
-	private final Supplier<MerkleMap<EntityNum, MerkleToken>> tokens;
+
 	private final TransactionalLedger<NftId, NftProperty, MerkleUniqueToken> nftsLedger;
 	private final TransactionalLedger<
 			Pair<AccountID, TokenID>,
 			TokenRelProperty,
 			MerkleTokenRelStatus> tokenRelsLedger;
+	private final TransactionalLedger<TokenID, TokenProperty, MerkleToken> tokensLedger;
```

Comment:
Here I would definitely suggest injecting the `BackingStore<TokenID, MerkleToken> tokens` instead of this ledger...

In general, I don't see a reason we need a "top-level" `TransactionalLedger` for tokens; only `MutableEntityAccess` needs to create an instance (using `BackingTokens`) that it can include in its `WorldLedgers`.